src/App.tsx

```diff
@@ -1,14 +1,67 @@
+/* eslint-disable no-alert */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
+import { Filter } from './enum/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [userId, setUserId] = useState(0);
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const [filter, setFilter] = useState(Filter.ALL);
+  const [query, setQuery] = useState('');
+
+  const newModel = (newTodo: Todo) => {
+    setUserId(newTodo.userId);
+
+    return setSelectedTodo(newTodo);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setTodos(data);
+      })
+      .catch(error => {
+        alert(error);
+      });
+  }, []);
+
+  const onReset = () => {
+    setUserId(0);
+    setSelectedTodo(null);
+  };
+
+  const filteredTodos = useMemo(() => {
+    if (filter === Filter.ALL) {
+      return [...todos];
+    }
+
+    if (filter === Filter.ACTIVE) {
+      return todos.filter(todo => !todo.completed);
+    }
+
+    if (filter === Filter.COMPLETED) {
+      return todos.filter(todo => todo.completed);
+    }
+
+    return [...todos];
```

Comment:
Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others